jina/jaml/__init__.py

```diff
@@ -49,17 +82,45 @@ class DummyClass:
     An expression can be any combination of literal values, references to a context, or functions.
     You can combine literals, context references, and functions using operators.
 
-    You need to use specific syntax to tell Jina to evaluate an expression rather than treat it as a string.
+    You need to use specific syntax to tell Jina to evaluate an expression rather than treat it as a string,
+    which is based on GitHub actions syntax, and looks like this:
 
     .. highlight:: yaml
     .. code-block:: yaml
 
         ${{ <expression> }}
 
-    To evaluate (i.e. substitute the value to the real value)
-    the expression when loading, use :meth:`load(substitute=True)`.
+    This expression can be evaluated directly (i.e. substituted by the real value) when being loaded,
+    by using :meth:`load(substitute=True)`
+
+    JAML supports three different kinds of variables to be used as expressions: `Environment variables`
+    (coming form the environment itself), `context variables` (being passed as a dict),
+    and `internal references` (included in the .yaml file itself).
+
+    An environment variable `var` is accessed through the following syntax:
+
+    .. highlight:: yaml
+    .. code-block:: yaml
+
+        ${{ env.var }}
+
+    Note the mandatory spaces before and after the variable denotation.
+
+    Context variables can be accessed using the following syntax:
+
+    .. highlight:: yaml
+    .. code-block:: yaml
+
+        ${{ context_var }}
+
+    Or, if you want to be explicit:
+
+    .. highlight:: yaml
+    .. code-block:: yaml
+
+        ${{ context.context_var }}
 
-    To substitute the value based on a dict,
+    These context variables are passed as a dict:
```

Comment:
I've always passed env vars directly, without the need to `ENV.` prefix. Will it not work from now on? In what situation would you want context variables and how would you pass them to a Flow?

Reply:
Passing env vars like this: ${{ var }} should never have worked in the first place, we believe. So now you either do ${{ env.var }} or $var; the latter is discouraged but should work perfectly fine. You should be able to pass a dict containing context variables to Flow.load_config(context={'context_var' : 1}) . Not sure what the use cases for context vars are, though.